library/common/extensions/retry/options/network_configuration/predicate.cc

```diff
@@ -32,12 +32,19 @@ NetworkConfigurationRetryOptionsPredicate::updateOptions(
     return Upstream::RetryOptionsPredicate::UpdateOptionsReturn{absl::nullopt};
   }
 
-  auto& extra_stream_info =
+  StreamInfo::ExtraStreamInfo* extra_stream_info =
       filter_state->getDataMutable<StreamInfo::ExtraStreamInfo>(StreamInfo::ExtraStreamInfo::key());
 
+  if (extra_stream_info == nullptr) {
```

Comment:
Nit:

```suggestion
  if (!extra_stream_info) {
```

Reply:
I did this to kick CI 😂 while we were having all those issues yesterday. I'll leave it in because I dont want to run CI again. But will update in a subsequent PR

Reply:
Not a problem to leave as is.